### PR TITLE
Make queries non-reactive by default

### PR DESCRIPTION
**Pull request by kostko**
I believe this should go into a new version tag (e.g. 0.2.0), since it is a breaking change.

### CODE REVIEW

**Comment by kostko** [comment]
This is ready for review.

File: src/api/resource.ts [hunk 1]
```diff
@@ -64,83 +71,99 @@ export abstract class Resource {
...
-            if (queryObserverId) {
-                // This query observer identifier has already been cached. Check if it exists and in this
-                // case just subscribe to all items.
-                let queryObserver = this.connection.queryObserverManager().get(queryObserverId, false);
-                if (queryObserver) {
-                    if (queryObserver.status === QueryObserverStatus.INITIALIZED ||
-                        queryObserver.status === QueryObserverStatus.REINITIALIZING) {
-                        subscriptions.push(queryObserver.observable().subscribe(observer));
-                    }
+            if (options.reactive) {
```
Comment (hadalin) [inline src/api/resource.ts]: Should this rather be `if (!options.reactive) return;` to not go one level deeper?

Comment (kostko) [inline src/api/resource.ts]: There is an `else` below.

Comment (hadalin) [inline src/api/resource.ts]: What about 
```ts
if (!options.reactive) {
   // Reactivity is disabled for this query.
   query = this.transformQuery(query);
   ...
}

 // Reactivity is enabled.
let queryObserverId = this._queryObserverIdCache[serializedQuery];
...
```

?

Comment (kostko) [inline src/api/resource.ts]: Sounds good, fixed.

File: src/api/resource.ts [hunk 2]
```diff
@@ -64,12 +71,28 @@ export abstract class Resource {
...
+            if (!options.reactive) {
+                // Reactivity is disabled for this query.
+                query = this.transformQuery(query);
+                this.connection.get(path, query).map((response: any) => {
+                    // Correctly handle paginated results.
+                    if (_.has(response, 'results')) return response.results;
+                    return response;
+                }).subscribe(observer);
+
+                return () => { /* No dispose function needed. */ };
```
Comment (lukaw3d) [inline src/api/resource.ts]: Should this call the `subscribe(observer).dispose()` in case dispose is called before server responds?

Comment (kostko) [inline src/api/resource.ts]: Good point, fixed.

**Pull request merged**